vms/platformvm/txs/executor/proposal_tx_executor.go

```diff
@@ -121,7 +126,7 @@ func (e *ProposalTxExecutor) AddValidatorTx(tx *txs.AddValidatorTx) error {
 
 	onAbortOuts, err := verifyAddValidatorTx(
 		e.Backend,
-		e.OnCommitState,
+		e.OnProposalBlockState,
```

Comment:
in all of these verify methods `OnProposalBlockState` should never be modified. Consider carving out of  `state.Chain` interface into a `state.ReadOnlyChain` containing only getters. It won't actually enforce any constness of proposalBlockState, but it'd be more expressive.
Deferred to a future PR to keep scope reduced.